src/journal/journalctl.c

```diff
@@ -577,6 +579,8 @@ static int parse_argv(int argc, char *argv[]) {
                         if (IN_SET(arg_output, OUTPUT_EXPORT, OUTPUT_JSON, OUTPUT_JSON_PRETTY, OUTPUT_JSON_SSE, OUTPUT_JSON_SEQ, OUTPUT_CAT))
                                 arg_quiet = true;
 
+                        if (OUTPUT_MODE_IS_JSON(arg_output))
+                                arg_json_format_flags = output_mode_to_json_format_flags(arg_output) | JSON_FORMAT_COLOR_AUTO;
```

Comment:
i think this should have an else branch that resets arg_json_format_flags back to JSON_FORMAT_OFF.

`journalctl -o json-pretty -o export` might otherwise produce unexpected output (i.e. I'd expect the second -o overrides the first in full, but without the else it would not)